### PR TITLE
Use context stack depth to set the Java stack size

### PR DESCRIPTION
**Pull request by Sabine-Schnabeltier**
Links the Java thread stack size to the requested `--context-stack-depth` to prevent Java stack overflow, and unifies headless and UI thread execution.

**Changes**

* **Thread Sizing:** Both headless and UI modes now execute inside a dedicated `TruffleSqueakVM-Thread`. The thread's stack size is allocated as `(contextStackDepth + 512) * 1024` bytes.
* **Centralized Constants:** Moves the default context depth (`2 << 12`) and introduces a new minimum depth (`128`) in `SqueakLanguageConfig.java`.
* **Input Clamping:** Clamps the context stack depth to the minimum in both `TruffleSqueakLauncher` and `SqueakOptions`.
* **Deprecation:** Removes `(0 = disabled)` from the help string since that is no longer supported.

**Results**

With this change, it is now possible to `update code from the server` using the default context stack depth of 8192. This was also tested with a variety of context stack depths down to the minimum of 128.

Also tested headless mode:
 
```
trufflesqueak % $TS_JVM_HOME/bin/trufflesqueak --download-image trufflesqueak --context-stack-depth 35000 -c "Time millisecondsToRun: [30000 factorial]"
[trufflesqueak] Running TruffleSqueak-25.0.1.image on Oracle GraalVM (latency mode)...
[trufflesqueak] Evaluating 'Time millisecondsToRun: [30000 factorial]'...
[trufflesqueak] Result: 986
```

_Note that using smaller context stack depths will result in more Truffle frame flushes and slower code execution._

### CODE REVIEW

**Comment by Sabine-Schnabeltier** [comment]
The second commit changes the command line parsing slightly:

```
trufflesqueak % $TS_JVM_HOME/bin/trufflesqueak --download-image trufflesqueak --experimental-options --context-stack-depth 35000 -c "Time millisecondsToRun: [30000 factorial]"
[trufflesqueak] Running TruffleSqueak-25.0.1.image on Oracle GraalVM (latency mode)...
[trufflesqueak] Evaluating 'Time millisecondsToRun: [30000 factorial]'...
[trufflesqueak] Result: 1002
```

File: src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java [hunk 1]
```diff
@@ -91,6 +95,19 @@ protected List<String> preprocessArguments(final List<String> arguments, final M
...
+                final String matchedPrefix = arg.startsWith(SqueakLanguageOptions.CONTEXT_STACK_DEPTH_FLAG)
+                                ? SqueakLanguageOptions.CONTEXT_STACK_DEPTH_FLAG
+                                : "--" + SqueakLanguageConfig.ID + "." + SqueakLanguageOptions.CONTEXT_STACK_DEPTH;
+
+                i = handleIntOption(arguments, i, arg, matchedPrefix, val -> {
+                    contextStackDepth = Math.max(SqueakLanguageConfig.MIN_CONTEXT_STACK_DEPTH, val);
+                }, unrecognized);
+
+                // Push the fully-qualified flag back into unrecognized.
+                unrecognized.add("--" + SqueakLanguageConfig.ID + "." + SqueakLanguageOptions.CONTEXT_STACK_DEPTH + "=" + contextStackDepth);
```
Comment (fniephaus) [inline src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java]: what is all this additional logic for? Why is the call to `handleIntOption()` not enough?

File: src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java [hunk 2]
```diff
@@ -71,7 +74,7 @@ public static SqueakContextOptions create(final OptionValues options) {
                             options.get(ResourceSummary),
                             options.get(Headless),
                             options.get(Interrupts),
-                            Math.max(1, options.get(ContextStackDepth)),
+                            Math.max(MIN_CONTEXT_STACK_DEPTH, options.get(ContextStackDepth)),
```
Comment (fniephaus) [inline src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java]: Is a min value really that useful here? Maybe we should have a min value for the thread stack size instead?

File: src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java (hunk 1 above)
Comment (Sabine-Schnabeltier) [reply src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java]: The extra code is to handle the language-qualified command line option: `--smalltalk.context-stack-depth=1000`

If handling that command line option form is not important, I can revert the last commit.

File: src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java (hunk 2 above)
Comment (Sabine-Schnabeltier) [reply src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java]: It was a pragmatic choice: the stack size is fine for all context stack depths, but small values cause a notifier when TruffleSqueak starts up:

`ts_run --experimental-options --context-stack-depth 16`

Screenshot 2026-07-12 at 13 36 20

I tried 16, 32, 64, 128; 128 was the first size where TruffleSqueak would behave normally.

I suppose having the larger minimum value may be concealing a hidden interpreter bug.

File: src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java (hunk 1 above)
Comment (fniephaus) [reply src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java]: Yeah, I don't think we need to handle language-qualified command line options. Could you revert the last commit please? Not quite sure why it dropped `contextBuilder.option(SqueakLanguageConfig.ID + "." + SqueakLanguageOptions.CONTEXT_STACK_DEPTH, Integer.toString(contextStackDepth));`

**Review by fniephaus — CHANGES_REQUESTED** [review]
Otherwise LGTM!

File: src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java (hunk 1 above)
Comment (Sabine-Schnabeltier) [reply src/de.hpi.swa.trufflesqueak.launcher/src/de/hpi/swa/trufflesqueak/launcher/TruffleSqueakLauncher.java]: Reverted!

File: src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java (hunk 2 above)
Comment (Sabine-Schnabeltier) [reply src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/SqueakOptions.java]: Reverted to minimum 1 frame since PR #291 takes care of the depth 128 limit.

**Review by fniephaus — APPROVED** [review]
LGTM, thank you!
